src/main/java/racingcar/domain/RandomNumberGenerator.java

```diff
@@ -0,0 +1,14 @@
+package racingcar.domain;
+
+import java.util.Random;
+
+public class RandomNumberGenerator implements NumberGenerator {
+
+    private static final int RANDOM_MAX = 10;
+
+    @Override
+    public int generate() {
```

Comment:
```suggestion
    public int generate(int max) {
```

`NumberGenerator` 는 숫자 생성에 대한 책임만 부여하면 어떨까요?
외부에서는 `RANDOM_MAX = 10;`에 대한 조건이 있는지 알기 힘들수 있다고 생각해요.

인자를 받도록 변경하면 이러한 장점들이 있을것 같아요.
- 테스트 용이성 : 이전 코드에서는 `RANDOM_MAX` 상수 값으로 고정되어 있었기 때문에 테스트 케이스 작성 시 입력값을 다양하게 넣어보기 어려웠습니다. 하지만 수정된 코드에서는 입력값으로 다양한 수를 전달하여 테스트가 가능해집니다.

- 오용 방지 : 항상 0부터 9까지의 숫자 중 하나를 반환했기 때문에, 이 메소드를 호출하는 쪽에서 예상치 못한 결과가 나올 수 있습니다. 하지만 인자를 주입받는다면 호출하는 쪽에서 원하는 최대값을 전달할 수 있으므로 오용 방지 또한 가능합니다.

- 확장성 : 항상 0부터 9까지의 숫자 중 하나만 반환할 수 있었으나, 인자를 주입받는다면 최대값을 매개변수로 받아서 그 범위 내의 임의 정수 값을 반환할 수 있으므로 더욱 확장성이 높아집니다.


---

해당내용과 관련있을것 같기도해서 공유해봅니다! 😄 
[테크 리더 3인이 말하는 "개발자 원칙](https://www.youtube.com/live/DJCmvzhFVOI?feature=share&t=1875)